cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -200,6 +201,26 @@ type APIEndpoint struct {
 	BindPort int32
 }
 
+// InitTimeouts holds timeout fields that relate to registering a new control-plane to the cluster,
+// via "kubeadm init"
+type InitTimeouts struct {
+
+	// ApiServerHealthCheck controls the timeout that we use for API server to appear
+	// +optional
+	ApiServerHealthCheck *metav1.Duration `json:"apiServerHealthCheck,omitempty"`
+}
+
+// JoinTimeouts holds timeout fields that relate to registering a new control-plane or node to the cluster,
+// via "kubeadm join
+type JoinTimeouts struct {
+	// ApiServerHealthCheck controls the timeout that we use for API server to appear
+	// +optional
+	ApiServerHealthCheck *metav1.Duration `json:"apiServerHealthCheck,omitempty"`
```

Comment:
internal types (kubeadm/types.go) should not have:
- JSON tags
- +optional

Reply:
OK, Thanks.